### ISSUE TITLE
Error in deploy Yunikorn on GCP

### ISSUE THREAD

**Issue by maprietom** [cloudera/yunikorn-core#18]
I'm trying to deploy a Yunikorn in a Kubernetes cluster in GCP. I followed the user-guide to do it, but when deploy the scheduler didn't work, I get this error on "yunikorn-scheduler-k8s":
/k8s_yunikorn_scheduler: line 1: ELF: not found
/k8s_yunikorn_scheduler: line 7: syntax error: unexpected ")"
I don't know where is the problem, Can I help me?
Thanks!!

**Comment by yangwwei** [comment]
Hi @maprietom 
Are you deploying on a GCE VM?  What's the OS version?
Not sure if this is related to our build options https://github.com/cloudera/yunikorn-k8shim/blob/5a799d31d08e628e6e1a253700fd854a8e23eb2d/Makefile#L69-L73

**Comment by sunilgovind** [comment]
Hi @maprietom 
Do you have any other log messages?

**Comment by dcaceres88** [comment]
Hi all guys,

I am a colleague of @maprietom and I try to explain more deeply the problem. 

We are trying to deploy Yunikorn over GKE using one of your examples (https://github.com/cloudera/yunikorn-k8shim/blob/master/deployments/scheduler/scheduler-load.yaml). Previously we are applied RBAC and create ConfigMap as you explain in your documentation. 
When we applied scheduler-load.yaml to our cluster, we receive this error:

"standard_init_linux.go:207: exec user process caused "no such file or directory"

Using Google we found similar issues in other projects that talk about a problem with the prompt we are using ( In that case we are using bash?). For that reason, @maprietom changed the prompt, and now our deployment uses sh instead of bash. Now, we have the error that explains @maprietom,  but the problem persists. 

Can you explain more about this behavior?.

Thanks in advance.

**Comment by dcaceres88** [comment]
Hi again, 

Finally, we have been able to deploy Yunikorn over GCP. We found a couple of problems to do that:

1- We modified Makefile file and adding "CGO_ENABLED=0" parameter into build_image method. Using Google we found similar problems, and with that change, we achieve "Turn off CGO since that can result in dynamic links to libc/libmusl.".  The result of this change is:

```
.PHONY: build_image
build_image: init
        @echo "building binary for scheduler docker image"
        CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
        go build -a -o=${RELEASE_BIN_DIR}/${BINARY} -ldflags \
        '-extldflags "-static" -X main.version=${VERSION} -X main.date=${DATE}' \
        -tags netgo -installsuffix netgo \
        ./pkg/scheduler/
```

Before the change, yunikorn-scheduler pod works ok. But when the process started, we received another error about "queue.yaml file is not present".  To fix that problem, we done number 2 change.

2- The YAML we used to deploy Yunikorn (https://github.com/cloudera/yunikorn-k8shim/blob/master/deployments/scheduler/scheduler-load.yaml) in our opinion is incomplete. We add volumeMounts parameter to complete the yaml. Now the appearance of scheduler-load.yaml is:


```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    app: yunikorn
  name: yunikorn-scheduler
spec:
  replicas: 1
  selector:
    matchLabels:
      app: yunikorn
  template:
    metadata:
      labels:
        app: yunikorn
      name: yunikorn-scheduler
    spec:
      containers:
        - name: yunikorn-scheduler-k8s
          image: yunikorn/yunikorn-scheduler-k8s:latest
          imagePullPolicy: IfNotPresent
		  volumeMounts:
            - name: config-volume
              mountPath: /etc/yunikorn/
          ports:
            - containerPort: 9080
        - name: yunikorn-scheduler-web
          image: yunikorn/yunikorn-web:latest
          imagePullPolicy: IfNotPresent
          ports:
            - containerPort: 9889
      volumes:
        - name: config-volume
          configMap:
            name: yunikorn-configs
---
apiVersion: v1
kind: Service
metadata:
  name: yunikorn-service
  labels:
    app: yunikorn-service
spec:
  ports:
    - name: yunikorn-service
      port: 9889
    - name: yunikorn-core
      port: 9080
  selector:
    app: yunikorn
  type: LoadBalancer

```

Now all of the pieces of deployment works ok. And we can start to deploy services over Yunikorn on GCP to proving all the solution. 

Thank you.

**Comment by yangwwei** [comment]
Hi @dcaceres88 

For the 1st issue, I just take a deeper look about `CGO_ENABLED` option, I think it makes sense to disable CGO since dynamical links may cause cross-platform issues like you guys met. Do you want to create a PR for this? Thank you! 

For the 2nd issue, it was fixed in this commit https://github.com/cloudera/yunikorn-k8shim/commit/5a799d31d08e628e6e1a253700fd854a8e23eb2d a couple of days ago.

**Comment by sunilgovind** [comment]
Thanks @maprietom for investigating.
It makes sense to disable CGO_ENABLED.  I tested locally and seems ok.

As @yangwwei  mentioned, it will be great if you could help to create PR and we can help to merge same as contributions from you. Thanks for trying this

**Comment by dcaceres88** [comment]
Hi guys, 

We opened the next PR (https://github.com/cloudera/yunikorn-k8shim/pull/13).

Thanks.

**Comment by yangwwei** [comment]
This is fixed by cloudera/yunikorn-k8shim#13
